src/main/java/edu/ie3/datamodel/utils/validation/ValidationUtils.java

```diff
@@ -354,38 +326,44 @@ public static <T> Predicate<T> distinctByKey(Function<? super T, ?> keyExtractor
   }
 
   /**
-   * Checks if the provided set of unique entities only contains elements with distinct UUIDs and
-   * either returns a string with duplicated UUIDs or an empty optional otherwise.
+   * Method to check for duplicate fields in a set of {@link UniqueEntity}.
    *
-   * @param entities the entities that should be checkd for UUID uniqueness
-   * @return either a string wrapped in an optional with duplicate UUIDs or an empty optional
+   * @param entities to be checked
+   * @param supplier for the field
+   * @return a list of {@link Try}.
+   * @param <A> type of the field
+   * @param <B> type of the {@link UniqueEntity}
    */
-  protected static Optional<String> checkForDuplicateUuids(Set<UniqueEntity> entities) {
-    if (distinctUuids(entities)) {
-      return Optional.empty();
-    }
-    String duplicationsString =
-        entities.stream()
-            .collect(Collectors.groupingBy(UniqueEntity::getUuid, Collectors.counting()))
-            .entrySet()
-            .stream()
-            .filter(entry -> entry.getValue() > 1)
-            .map(
-                entry -> {
-                  String duplicateEntitiesString =
-                      entities.stream()
-                          .filter(entity -> entity.getUuid().equals(entry.getKey()))
-                          .map(UniqueEntity::toString)
-                          .collect(Collectors.joining("\n - "));
-
-                  return entry.getKey()
-                      + ": "
-                      + entry.getValue()
-                      + "\n - "
-                      + duplicateEntitiesString;
-                })
-            .collect(Collectors.joining("\n\n"));
-
-    return Optional.of(duplicationsString);
+  protected static <A, B extends UniqueEntity>
+      List<Try<Void, DuplicateEntitiesException>> checkForDuplicate(
+          Collection<B> entities, FieldSupplier<A, B> supplier) {
+    Map<A, List<B>> duplicates =
+        entities.stream().collect(Collectors.groupingBy(supplier::getField));
+
+    List<Try<Void, DuplicateEntitiesException>> exceptions = new ArrayList<>();
+
+    duplicates.entrySet().stream()
+        .filter(e -> e.getValue().size() > 1)
+        .forEach(
+            duplicate -> {
+              A key = duplicate.getKey();
+              exceptions.add(
+                  Failure.ofVoid(
+                      new DuplicateEntitiesException(
+                          key.getClass().getSimpleName(), duplicates.get(key))));
+            });
+
+    return exceptions;
```

Comment:
Love everything about the way you arranged this, just the `map` operation that was used before, I think is more neat.

This way, you don't need the intermediary `exceptions` list.

I tried it here, let me know what you think:

```suggestion
    return entities.stream().collect(Collectors.groupingBy(supplier::getField)).entrySet().stream()
        .filter(e -> e.getValue().size() > 1)
        .map(
            duplicate ->
                Failure.ofVoid(
                    new DuplicateEntitiesException(
                        duplicate.getKey().getClass().getSimpleName(), duplicate.getValue())))
        .collect(Collectors.toList());
```